### PR TITLE
IGAPP-399: Format dates in right locales

### LINKED ISSUE #1384: IGAPP-399: Remove date formatting workaround and format it in right locale again
# steffenkleinle - 25.11.2020, 14:13:40

Currently there is a workaround at several places in the native app and the api-client to circumvent the following bug: https://sentry.tuerantuer.org/organizations/digitalfabrik/issues/18/?query=is%3Aunresolved

The workaround is the following: Always use 'en' as locale in which the dates are formatted.

Make sure the app does not crash anymore. Possibly this should be done with the removal of moment.js in the following issue: https://issues.tuerantuer.org/browse/IGAPP-281

Also make sure that our default locale is used if the locale is not available (https://moment.github.io/luxon/docs/manual/intl.html#default-locale).

Check if refactoring makes sense and do it or create a new issue for it.

Testing information: Check if the last update timestamps are correctly formatted and localized in all routes (native + web). Check that start/end times for events are correctly formatted and localized.

**Linked issues:**
- Blocks: IGAPP-281
- Relates: IGAPP-281
- Relates: IGAPP-376

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Fixes #1384.

### CODE REVIEW

**Comment by steffenkleinle** [comment]
> I did get an error in iOS though: Error: Invalid DateTime: unparsable: the input "undefined" can't be parsed as ISO 8601

Where exactly did you get that error? Some specific event, page, ...?

**Comment by LeandraH** [comment]
Sorry, whenever opening a new city

**Comment by steffenkleinle** [comment]
> Sorry, whenever opening a new city

Not a new issue introduced in my PR, created a new issue: #2314

**Review by sarahsporck — CHANGES_REQUESTED** [review]
Tested in Firefox and on ios simulator.
On native the page timestamp is not yet correct. And sometimes I randomly get an error  [Error: Invalid DateTime: unparsable: the input "undefined" can't be parsed as ISO 8601]. But not sure where this comes from.

File: api-client/src/models/__tests__/__snapshots__/DateModel.spec.ts.snap [hunk 1]
```diff
@@ -1,67 +1,56 @@
...
-
 exports[`DateModel toTimeSpanString() should return only start date + end date if allDay is true 1`] = `
-Array [
+[
   "de: 27. November 2017 - 28. November 2017",
   "en: November 27, 2017 - November 28, 2017",
   "fr: 27 novembre 2017 - 28 novembre 2017",
   "ar: ٢٧ نوفمبر ٢٠١٧ - ٢٨ نوفمبر ٢٠١٧",
-  "fa: ۲۷ نوامبر ۲۰۱۷ - ۲۸ نوامبر ۲۰۱۷",
+  "fa: ۶ آذر ۱۳۹۶ - ۷ آذر ۱۳۹۶",
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply api-client/src/models/__tests__/__snapshots__/DateModel.spec.ts.snap:9]: I am guessing that the other localizations were the same for moment as well.

File: api-client/src/models/DateModel.ts [hunk 2]
```diff
@@ -25,46 +23,20 @@ class DateModel {
...
-        // if allDay: only date, else: date + time
-        span += this._allDay
-          ? formatter.format(this._endDate, {
-              format: 'DDD',
-            })
-          : formatter.format(this._endDate, {
-              format: 'DDD t',
-            })
-      }
+    if (this.startDate.hasSame(this.endDate, 'day')) {
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply api-client/src/models/DateModel.ts:35]: Actually, it should work. From the docs:
> d1.hasSame(d2, 'day');    // both DateTimes have the same calendar day (which implies they also have the same calendar year and month)

https://moment.github.io/luxon/#/math?id=comparing-datetimes

**Review by sarahsporck — APPROVED** [review]
Tested again and now works fine. As always I forgot that lastUpdate is not controlled by the app language.

**Pull request merged**